projects/packages/jetpack-mu-wpcom/src/features/wpcom-themes/includes/class-wpcom-themes-api.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Class WPCom_Themes_Api.
+ * Retrieves themes from the WordPress.com themes API.
+ *
+ * @package wpcom-themes
+ */
+
+/**
+ * Fetches themes from the WordPress.com themes API.
+ */
+class WPCom_Themes_Api {
+
+	/**
+	 * The URL of the WordPress.com themes API.
+	 */
+	const WP_COM_THEMES_API_URL = 'https://public-api.wordpress.com/rest/v1.2/themes?http_envelope=1&page=1&number=1000';
+
+	/**
+	 * The URL of the WordPress.com theme API.
+	 */
+	const WP_COM_THEME_API_URL = 'https://public-api.wordpress.com/rest/v1.2/themes/%s?http_envelope=1';
+
+	/**
+	 * Cache handler.
+	 *
+	 * @var WPCom_Themes_Cache $cache
+	 */
+	private WPCom_Themes_Cache $cache;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param WPCom_Themes_Cache $cache Cache handler.
+	 */
+	public function __construct( WPCom_Themes_Cache $cache ) {
+		$this->cache = $cache;
+	}
+
+	/**
+	 * Returns an array of themes fetched from the WordPress.com themes API.
+	 * Caching is used to avoid fetching the themes on every request.
+	 *
+	 * @param string $cache_key Key of the cache where the API response will be cached.
+	 * @param array  $params    Query params to pass to the API URL.
+	 *
+	 * @return array<stdClass> An array with all the WPCom themes.
+	 */
+	protected function fetch_themes( string $cache_key, array $params = array() ): array {
```

Comment:
This function calls the API through HTTP. On Simple sites we can call the code directly, which means that we don't need to wait for an HTTP request. I think we can do this using Client::wpcom_json_api_request_as_blog

Reply:
Done in [458df77](https://github.com/Automattic/jetpack/pull/41374/commits/458df77a84a7c2a51b44c19e5e2e999b0bbdad31) - we should now be able to modify the API and see the results when testing.

Reply:
I reverted this because it was broken, but it should give you an idea how to do it!